api/v1alpha1/dnspolicy_types.go

```diff
@@ -239,37 +226,10 @@ func (l *DNSPolicyList) GetItems() []kuadrant.Policy {
 // By default, this health check will be applied to each unique DNS A Record for
 // the listeners assigned to the target gateway
 type HealthCheckSpec struct {
-	Endpoint                  string           `json:"endpoint,omitempty"`
-	Port                      *int             `json:"port,omitempty"`
-	Protocol                  *string          `json:"protocol,omitempty"`
-	FailureThreshold          *int             `json:"failureThreshold,omitempty"`
-	AdditionalHeadersRef      *string          `json:"additionalHeadersRef,omitempty"`
-	ExpectedResponses         []int            `json:"expectedResponses,omitempty"`
-	AllowInsecureCertificates bool             `json:"allowInsecureCertificates,omitempty"`
-	Interval                  *metav1.Duration `json:"interval,omitempty"`
-}
-
-func (s *HealthCheckSpec) Validate() error {
-	if s.Interval != nil {
-		if s.Interval.Duration < (time.Second * 5) {
-			return fmt.Errorf("invalid value for spec.healthCheckSpec.interval %v, it cannot be shorter than 5s", s.Interval.Duration)
-		}
-	}
-
-	return nil
-}
-
-func (s *HealthCheckSpec) Default() {
-	if s.Interval == nil {
-		s.Interval = &metav1.Duration{
-			Duration: time.Second * 30,
-		}
-	}
-
-	if s.Protocol == nil {
-		protocol := "HTTPS"
-		s.Protocol = &protocol
-	}
+	Endpoint         string `json:"endpoint"`
+	Port             int    `json:"port"`
+	Protocol         string `json:"protocol"`
+	FailureThreshold int    `json:"failureThreshold"`
```

Comment:
You might want to consider adding some CEL validations for these, presumably there are values that we should just not accept for some of these.

Reply:
Also, a description of each would be good since it will end up as the description in the generated CRD.

Reply:
nice, added.